core/src/std_packages.rs

```diff
@@ -176,44 +289,172 @@ pub fn native_block_content<T, U>(
     Ok(Element::Compound(elements))
 }
 
-/// Example function for pushing a string to a list
-pub fn native_push_list<T, U>(
+/// Helper function to create text elements
+fn text_element(contents: String, id: GranularId) -> Element {
+    Element::Module {
+        name: "__text".to_string(),
+        args: Default::default(),
+        body: contents,
+        inline: true,
+        id: id,
+    }
+}
+
+/// Declare a constant
+pub fn const_decl<T, U>(
     ctx: &mut Context<T, U>,
-    body: &str,
+    value: &str,
     args: HashMap<String, ArgValue>,
-    _inline: bool,
-    _output_format: &OutputFormat,
-    _id: &GranularId,
+    _: bool,
+    _: &OutputFormat,
+    _: &GranularId,
 ) -> Result<Element, CoreError> {
-    let key = args.get("key").unwrap().as_str().unwrap();
-    let entry = ctx.lists.entry(key.to_string()).or_default();
-    entry.push(body.to_string());
+    let name = args.get("name").unwrap().as_str().unwrap();
+    ctx.state.variables.constant_declare(name, value)?;
+
     Ok(Element::Compound(vec![]))
 }
 
-/// Example function for reading a list
-pub fn native_read_list<T, U>(
+/// Read a constant
+pub fn const_read<T, U>(
     ctx: &mut Context<T, U>,
-    _body: &str,
+    _: &str,
     args: HashMap<String, ArgValue>,
-    _inline: bool,
-    _output_format: &OutputFormat,
+    _: bool,
+    format: &OutputFormat,
     id: &GranularId,
 ) -> Result<Element, CoreError> {
-    let key = args.get("key").unwrap().as_str().unwrap();
-    if let Some(list) = ctx.lists.get(key) {
-        Ok(Element::Module {
-            name: "__text".to_string(),
-            args: Default::default(),
-            body: list.join(","),
-            inline: false,
-            id: id.clone(),
-        })
-    } else {
-        Ok(Element::Compound(vec![]))
+    let name = args.get("name").unwrap().as_str().unwrap();
+
+    match ctx.state.variables.get(name) {
+        Some(value @ variables::Value::Constant(_)) => {
+            Ok(text_element(value.to_string(), id.clone()))
+        }
+        Some(value) => Err(CoreError::TypeMismatch {
+            name: name.to_string(),
+            expected_type: VarType::Constant,
+            present_type: value.get_type(),
+        }),
+        None => {
+            let default = args.get("default").map(ArgValue::as_str).flatten().unwrap();
+            if default == "<log error>" {
+                // No default was provided as fallback, let's log a error
+                log_var_error(ctx, name, "const-read", format);
```

Comment:
This should generate an `[error]` node to facilitate inline error reporting and keep our convention